test/1_stdlib/subString.swift

```diff
@@ -1,15 +1,110 @@
-// RUN: %target-run-simple-swift | FileCheck %s
+// RUN: rm -rf %t
+// RUN: mkdir -p %t
+// RUN: %target-build-swift %s -o %t/a.out
+// RUN: %target-run %t/a.out
 // REQUIRES: executable_test
 
-func test(_ s: String) {
-  print(s)
-  var s2 = s[s.index(s.startIndex, offsetBy: 2)..<s.index(s.startIndex, offsetBy: 4)]
-  print(s2)
-  var s3 = s2[s2.startIndex..<s2.startIndex]
-  var s4 = s3[s2.startIndex..<s2.startIndex]
+import StdlibUnittest
+
+var SubstringTests = TestSuite("SubstringTests")
+
+func checkMatch<S: Collection, T: Collection
+  where S.Index == T.Index, S.Iterator.Element == T.Iterator.Element,
+  S.Iterator.Element: Equatable>(
+  _ x: S, _ y: T, _ i: S.Index) {
+  
+  expectEqual(x[i], y[i])
+}
+
+let s = "abcdefg"
+
+SubstringTests.test("String") {
+  let s1 = s[s.index(s.startIndex, offsetBy: 2) ..<
+    s.index(s.startIndex, offsetBy: 4)]
+  let s2 = s1[s1.startIndex..<s1.endIndex]
+  let s3 = s2[s1.startIndex..<s1.endIndex]
+  
+  expectEqual(s1, "cd")
+  expectEqual(s2, "cd")
+  expectEqual(s3, "cd")
+}
+
+SubstringTests.test("CharacterView")
+  .xfail(.always("CharacterView slices don't share indices"))
+  .code {
+  var t = s.characters.dropFirst(2)
+  var u = t.dropFirst(2)
+  
+  checkMatch(s.characters, t, t.startIndex)
+  checkMatch(s.characters, t, t.index(after: t.startIndex))
+  checkMatch(s.characters, t, t.index(before: t.endIndex))
+  
+  t.replaceSubrange(t.startIndex...t.startIndex, with: ["C"])
+  expectEqual(String(t), "Cdefg")
+  expectEqual(s, "abcdefg")
+  
+  checkMatch(s.characters, t, u.startIndex)
```

Comment:
I don't think this is a valid assertion.  `t` has been mutated before the position referenced by `u.startIndex`